shell/platform/android/android_surface_gl.cc

```diff
@@ -92,6 +92,9 @@ bool AndroidSurfaceGL::SetNativeWindow(
   FML_DCHECK(IsValid());
   FML_DCHECK(window);
   native_window_ = window;
+  // Ensure the destructor is called since it destroys the `EGLSurface` before
+  // creating a new onscreen surface.
+  onscreen_surface_ = nullptr;
```

Comment:
Is this for memory optimization?  I think `onscreen_surface_.reset()` would be better so readers know we are dealing with a smart pointer.

Reply:
We are already applying this optimization today.